core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -84,23 +85,30 @@ private[kafka] object ConsumerStage {
           override def groupId: String = _groupId
           lazy val committer: Committer = {
             val ec = materializer.executionContext
-            new KafkaAsyncConsumerCommitterRef(consumer, commitTimeout)(ec)
+            KafkaAsyncConsumerCommitterRef(consumer, commitTimeout)(ec)
           }
         }
     }
   }
 
   // This should be case class to be comparable based on ref and timeout. This comparison is used in CommittableOffsetBatchImpl
-  case class KafkaAsyncConsumerCommitterRef(ref: ActorRef, timeout: FiniteDuration)(implicit ec: ExecutionContext) extends Committer {
+  case class KafkaAsyncConsumerCommitterRef(ref: ActorRef, commitTimeout: FiniteDuration)(implicit ec: ExecutionContext) extends Committer {
     import scala.collection.breakOut
     import akka.pattern.ask
-    implicit val to = Timeout(timeout)
+    implicit val to = Timeout(commitTimeout)
+    private val commitTimeoutException = new CommitTimeoutException(s"Kafka commit took longer than: $commitTimeout")
```

Comment:
Nice to save the allocation, but, this will give a misleading stack trace, which isn't very nice.

Reply:
:+1: